app/models/alchemy/page/page_elements.rb

```diff
@@ -134,6 +134,18 @@ def available_element_names
       @_available_element_names ||= available_element_definitions.map { |e| e['name'] }
     end
 
+    # Available element definitions excluding nested unique elements.
+    #
+    def available_elements_within_current_scope(parent)
+      @_available_elements = if parent
+        available_element_definitions(parent.name).reject do |e|
+          parent.nested_elements.select(&:unique).map(&:name).include? e["name"]
```

Comment:
Just see that we should store the parent unique element names in a variable so that we can spare selecting them on each iteration of `available_element_definitions` and since unique and name are database columns we can use a more efficient method to get the names:

```ruby
...
parents_unique_nested_elements = parent.nested_elements.where(unique: true).pluck(:name)
available_element_definitions(parent.name).reject do |e|
  parents_unique_nested_elements.include? e['name']
end
...
```